sherpa/utils/__init__.py

```diff
@@ -2736,6 +2737,56 @@ def parallel_map(function, sequence, numcores=None):
 
     return run_tasks(procs, err_q, out_q, numcores)
 
+def parallel_map_funcs(funcs, datasets, numcores=None):
+
+    if not numpy.iterable(funcs):
+        raise TypeError("input '%s' is not iterable" % repr(funcs))
+
+    if not numpy.iterable(datasets):
+        raise TypeError("input '%s' is not iterable" % repr(datasets))
+
+    for func in funcs:
+        if not callable(func):
+            raise TypeError("input func '%s' is not callable" % repr(func))
+
+    funcs_size = len(funcs)
+    datasets_size = len(datasets)
+    if funcs_size != datasets_size:
+        msg = "input funcs (%d) and datsets (%d) size must be same" % \
+            (funcs_size, datasets_size)
+        raise TypeError(msg)
+
+    if not _multi or datasets_size == 1 or \
+            (numcores is not None and numcores < 2):
+        return list(map(funcs[0], datasets))
+
+    if numcores is None:
+        numcores = _ncpus
+
+    # Returns a started SyncManager object which can be used for sharing
+    # objects between processes. The returned manager object corresponds
+    # to a spawned child process and has methods which will create shared
+    # objects and return corresponding proxies.
+    manager = multiprocessing.Manager()
+
+    # Create FIFO queue and lock shared objects and return proxies to them.
+    # The managers handles a server process that manages shared objects that
+    # each slave process has access to.  Bottom line -- thread-safe.
+    out_q = manager.Queue()
+    err_q = manager.Queue()
+    lock = manager.Lock()
+
+    # if datasets is less than numcores, only use len datasets number of
+    # processes
+    numcores = min(funcs_size, datasets_size)
+
+    procs = []
+    for ii, chunk in enumerate(datasets):
```

Comment:
`for func, chunk in zip(funcs, datasets):` is functionally the same but makes more clear what's going on here.